### PR TITLE
Iht 4276 Extra error handling in PDF functionality and refactor for readability and removal of duplicated code

### PR DESCRIPTION
**Pull request by grantrobson**

### CODE REVIEW

File: app/iht/utils/pdf/XmlFoToPDF.scala [hunk 1]
```diff
@@ -28,11 +29,15 @@ import iht.utils.CommonHelper
 import iht.utils.tnrb.TnrbHelper
 import iht.utils.xml.ModelToXMLSource
 import models.des.iht_return.IHTReturn
-import org.apache.fop.apps._
+import org.apache.fop.apps.{FOUserAgent, _}
```
Comment (dmbeer) [inline app/iht/utils/pdf/XmlFoToPDF.scala]: Change Import statement to either list specific classes or just _ for all

File: app/iht/controllers/application/pdf/PDFController.scala [hunk 2]
```diff
@@ -48,11 +48,11 @@ trait PDFController extends ApplicationController with IhtActions {
...
 
-  def onPDFSummary = authorisedForIht {
+  def onPreSubmissionPDF = authorisedForIht {
     implicit user => implicit request => {
       Logger.info("Generating Summary PDF")
       withApplicationDetails { regDetails => applicationDetails =>
-        val pdfByteArray = xmlFoToPDF.generateSummaryPDF(
+        val pdfByteArray = xmlFoToPDF.createPreSubmissionPDF(
           regDetails,
           applicationDetails,
```
Comment (dmbeer) [inline app/iht/controllers/application/pdf/PDFController.scala]: watch formatting and unnecessary line breaks

File: app/iht/utils/pdf/XmlFoToPDF.scala [hunk 3]
```diff
@@ -42,52 +47,79 @@ object XmlFoToPDF extends XmlFoToPDF
...
+      Play.classloader.getResourceAsStream(filePathForClearanceXSL))
     val transformerFactory: TransformerFactory = TransformerFactory.newInstance
     transformerFactory.setURIResolver(new StylesheetResolver)
-
     val transformer: Transformer = transformerFactory.newTransformer(template)
+    setupTransformerEventHandling(transformer)
 
-    val fop: Fop = fopFactory.newFop(MimeConstants.MIME_PDF, foUserAgent, pdfoutStream)
-    val res = new SAXResult(fop.getDefaultHandler)
+    transformer.setParameter("pdfFormatter", PdfFormatter)
```
Comment (dmbeer) [inline app/iht/utils/pdf/XmlFoToPDF.scala]: can we possibly make some of the common parameters into a common method and set them that way

Comment (grantrobson) [inline app/iht/utils/pdf/XmlFoToPDF.scala]: Done. Please review

**Pull request merged**